easier_net/fit_easier_net.py

```diff
@@ -150,109 +142,40 @@ def main(args=sys.argv[1:]):
     x = dataset_dict["x"]
     y = dataset_dict["y"]
 
+#TODO: remove this + args in base estimator upon confirmation
     n_inputs = x.shape[1]
     n_out = 1 if args.num_classes == 0 else args.num_classes
     n_obs = x.shape[0]
 
-    # """
-    # Bootstrap sample
-    # """
-    # if args.bootstrap:
-    #     not_same_uniq_classes = True
-    #     while not_same_uniq_classes:
-    #         chosen_idxs = np.random.choice(n_obs, size=n_obs, replace=True)
-    #         x = x[chosen_idxs]
-    #         y = orig_y.flatten()[chosen_idxs].reshape((-1, 1))
-    #         if args.num_classes >= 2:
-    #             not_same_uniq_classes = np.unique(y).size != np.unique(orig_y).size
-    #         else:
-    #             not_same_uniq_classes = False
-    # else:
-    #     y = orig_y 
-
     """
     Fit EASIER-net
     """
     print("Fitting EASIER-net")
-    base_estimator = sier_net.SierNetEstimator(
-        n_inputs=n_inputs,
+    base_estimator = easier_net.EasierNetEstimator(
+        # n_inputs=n_inputs,
         input_filter_layer=args.input_filter_layer,
         n_layers=args.n_layers,
         n_hidden=args.n_hidden,
-        n_out=n_out,
+        # n_out=n_out,
         full_tree_pen=args.full_tree_pen,
         input_pen=args.input_pen,
-        batch_size=(n_obs // args.num_batches + 1),
+        num_batches=args.num_batches,
+        # batch_size=(n_obs // args.num_batches + 1),
         num_classes=args.num_classes,
         max_iters=args.max_iters,
         max_prox_iters=args.max_prox_iters,
-        # Weight classes by inverse of their observed ratios. Trying to balance classes
-        weight=n_obs / (args.num_classes * np.bincount(y.flatten()))
-        if args.num_classes >= 2
-        else None,
     )
-    # if args.fold_idxs_file is not None:
-    #     with open(args.fold_idxs_file, "rb") as f:
-    #         fold_idx_dict = pickle.load(f)
-    #         num_folds = len(fold_idx_dict)
 
-    #     parallel = Parallel(n_jobs=args.n_jobs, verbose=True, pre_dispatch=args.n_jobs)
-    #     all_estimators = parallel(
-    #         delayed(_fit)(
-    #             base_estimator, #single sier net
-    #             x,
-    #             y,
-    #             train=fold_idx_dict[fold_idx]["train"],
-    #             max_iters=args.max_iters,
-    #             max_prox_iters=args.max_prox_iters,
-    #             seed=args.seed + num_folds * init_idx + fold_idx,
-    #         )
-    #         for fold_idx, init_idx in itertools.product(
-    #             range(num_folds), range(args.num_inits)
-    #         )
-    #     )
-
-    #     # Just printing things from the first fold
-    #     logging.info(f"sample estimator 0 fold 0")
-    #     all_estimators[0].net.get_net_struct()
-
-    #     assert (num_folds * args.num_inits) == len(all_estimators)
+    all_estimators = base_estimator.fit(
+        x[np.arange(x.shape[0])], y[np.arange(x.shape[0])]
```

Comment:
hm? was this for testing? cause it should just say `x, y` (what the code is currently saying is ultimately equivalent, just lots for code)